File: 1-100/twoSum/twoSum.go
package twoSum

import "fmt"

func main() {
	var nums = [...]int{2, 7, 11, 15}
	var (
		target = 9
	)
	fmt.Printf("%d\n", twoSum(nums[:], target))
	fmt.Println("Hello, World!")
	//fmt.Println("",_,ok=nums[], target)
}

func twoSum(nums []int, target int) []int {
	var mp = make(map[int]int)
	var res []int
	for i := 0; i < len(nums); i++ {
		var dif = target - nums[i]
		if val, ok := mp[dif]; ok {
			return append(res, val, i)
		} else {
			mp[nums[i]] = i
		}
	}
	return append(res, 0, 0)
}

File: 1-100/twoSum/twoSum.rs
use std::arch::x86_64::_xsaves;
use std::collections::HashMap;

fn main() {
    let target: i32 = 6;
    let nums = vec![3,3];
    for i in twoSum(nums, target) {
        println!("{}", i);
    }

    // println!("Hello, world!");
}

fn twoSum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut map = HashMap::new();
    for i in 0..nums.len() {
        let tag = (target-nums[i]);
        match map.get_key_value(&tag) {
            Some((_,j)) => return vec![*j as i32, i as i32],
            None => {map.insert(nums[i], i);}
        }
    }
    return vec![];
}